src/ipc/ipc-common/IPCProxyBase.h

```diff
@@ -43,23 +42,132 @@
 
 namespace facelift {
 
-template<typename AdapterType>
-class IPCProxyBase : public AdapterType, protected IPCProxyBaseBase
+class Monitor;
+class IObserver
+{
+public:
+    virtual ~IObserver()=default;
+    virtual void handle_event(Monitor & source) = 0;
+};
+// Monitor
+class Monitor
+{
+    bool m_ready{};
+    std::vector<IObserver *> m_observers;
+public:
+    Monitor()=default;
+    virtual ~Monitor()=default;
+
+    void add(IObserver& observer){
+        m_observers.push_back(&observer);
+    }
+    void del(IObserver& observer){
+        m_observers.erase( remove(m_observers.begin(), m_observers.end(), &observer), m_observers.end() );
+    }
+    void notify(Monitor& source ){
+        for (auto observer: m_observers){
+            observer->handle_event(source);
+        }
+    }
+
+    bool stateReady() const { return m_ready; }
+    void stateReadyChanged(bool val)
+    {
+        this->m_ready = val;
+        notify(*this);
+    }
+};
+// IsReadyObserver
+class IsReadyObserver : public QObject
+{
+    Q_OBJECT
+    bool m_ready{};
```

Comment:
{} is not needed here, however it would be nice to have it initialized explicitly as false